src/Mod/Draft/draftgeoutils/geometry.py

```diff
@@ -275,32 +262,119 @@ def getRotation(v1, v2=App.Vector(0, 0, 1)):
     return App.Rotation(axis, angle)
 
 
-def isPlanar(shape):
+def isPlanar(shape, tol=-1):
     """Return True if the given shape or list of points is planar."""
-    n = getNormal(shape)
-    if not n:
-        return False
 
+    # for points    
     if isinstance(shape, list):
         if len(shape) <= 3:
             return True
         else:
-            for v in shape[3:]:
-                pv = v.sub(shape[0])
-                rv = DraftVecUtils.project(pv, n)
-                if not DraftVecUtils.isNull(rv):
-                    return False
+            poly = Part.makePolygon(shape)
+            if isStraightLine(poly, tol):
+                return True
+            plane = poly.findPlane(tol)
+            if plane:
+                return True
+            else:
+                return False
+    
+    # for shapes
+    # because Part.Shape.findPlane return None for Vertex and straight edges
+    if shape.ShapeType == "Vertex":
+        return True
+
+    if isStraightLine(shape, tol):
+        return True
+
+    plane = shape.findPlane(tol)
+    if plane:
+        return True
     else:
-        if len(shape.Vertexes) <= 3:
-            return True
+        return False
+
+
+def isStraightLine(shape, tol=-1):
+    """Return True if shape is a straight line."""
+    # function used in other methods because Part.Shape.findPlane assign a
+    # plane and normal to straight wires creating priviliged directions
+    # and to deal with straight wires with overlapped edges 
+    
+#    if hasattr(shape,"Faces"):
+    if len(shape.Faces) != 0:
+        return False
 
-        for p in shape.Vertexes[1:]:
-            pv = p.Point.sub(shape.Vertexes[0].Point)
-            rv = DraftVecUtils.project(pv, n)
-            if not DraftVecUtils.isNull(rv):
+    if len(shape.Edges) == 0:
+        return False
+
+    if len(shape.Edges) >= 1:
+        startEdge = shape.Edges[0]
+        dirStartEdge = startEdge.tangentAt(startEdge.FirstParameter)
+        err = shape.globalTolerance(tol)
+        for edge in shape.Edges:
+            firstPoint = edge.firstVertex().Point
+            lastPoint = edge.lastVertex().Point
+            dirEdge = edge.tangentAt(edge.FirstParameter)
+            # chek if edge is curve or no parallel to startEdge
+            # because sin(x) = x + O(x**3), for small angular deflection it's
+            # enough use the cross product of directions (or dot with a normal)
+            if (abs(edge.Length - firstPoint.distanceToPoint(lastPoint)) > err
+               or abs(dirStartEdge.cross(dirEdge).Length) > err):
```

Comment:
Normally the next line in parentheses must align with the contents of the previous line.
```py
if (abs(edge.Length ...) > err
    or abs(...) > err):
```